apps/addons/tests/test_models.py

```diff
@@ -2338,35 +2338,66 @@ def cleanup(self):
         self.empty_index('default')
 
     def test_no_addons(self):
-        eq_(Addon.search().count(), 0)
+        eq_(Addon.search_public().count(), 0)
 
     def test_create(self):
-        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo')
+        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo',
+                                     status=amo.STATUS_PUBLIC)
         self.refresh()
-        eq_(Addon.search().count(), 1)
-        eq_(Addon.search().query(name='woo')[0].id, addon.id)
+        eq_(Addon.search_public().count(), 1)
+        eq_(Addon.search_public().query(name='woo')[0].id, addon.id)
 
     def test_update(self):
-        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo')
+        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo',
+                                     status=amo.STATUS_PUBLIC)
         self.refresh()
-        eq_(Addon.search().count(), 1)
+        eq_(Addon.search_public().count(), 1)
 
         addon.name = 'yeah'
         addon.save()
         self.refresh()
 
-        eq_(Addon.search().count(), 1)
-        eq_(Addon.search().query(name='woo').count(), 0)
-        eq_(Addon.search().query(name='yeah')[0].id, addon.id)
+        eq_(Addon.search_public().count(), 1)
+        eq_(Addon.search_public().query(name='woo').count(), 0)
+        eq_(Addon.search_public().query(name='yeah')[0].id, addon.id)
+
+    def test_user_disable(self):
+        """Test that add-ons are removed from search results after being
+        disabled by their developers."""
+        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo',
+                                     status=amo.STATUS_PUBLIC)
+        self.refresh()
+        eq_(Addon.search_public().count(), 1)
+
+        addon.disabled_by_user = True
+        addon.save()
+        self.refresh()
+
+        eq_(Addon.search_public().count(), 0)
+
+    def test_user_switch_to_unlisted(self):
+        """Test that add-ons are removed from search results after being
+        switched to unlisted."""
+        addon = Addon.objects.create(type=amo.ADDON_EXTENSION, name='woo',
+                                     status=amo.STATUS_PUBLIC)
+        self.refresh()
+        eq_(Addon.search_public().count(), 1)
+
+        addon.is_listed = False
+        addon.save()
+        self.refresh()
+
+        eq_(Addon.search_public().count(), 0)
```

Comment:
Yeah, makes sense. I was assuming we already had one, but it looks like we don't.